google/cloud/google_cloud_cpp_common.cmake

```diff
@@ -214,6 +214,34 @@ install(
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/google_cloud_cpp_common"
     COMPONENT google_cloud_cpp_development)
 
+# Create a header-only library for the mocks. We use a CMake `INTERFACE` library
+# for these, a regular library would not work on macOS (where the library needs
+# at least one .o file). Unfortunately INTERFACE libraries are a bit weird in
+# that they need absolute paths for their sources.
+set(relative_mock_files mocks/mock_stream_range.h)
+set(mock_files)
+foreach (file IN LISTS relative_mock_files)
+    list(APPEND mock_files "${CMAKE_CURRENT_SOURCE_DIR}/${file}")
+endforeach ()
+add_library(google_cloud_cpp_mocks INTERFACE)
+target_sources(google_cloud_cpp_mocks INTERFACE ${mock_files})
+target_link_libraries(google_cloud_cpp_mocks INTERFACE google-cloud-cpp::common)
+set_target_properties(google_cloud_cpp_mocks PROPERTIES EXPORT_NAME
+                                                        google-cloud-cpp::mocks)
+target_include_directories(
+    google_cloud_cpp_mocks
+    INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+              $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
+              $<INSTALL_INTERFACE:include>)
+target_compile_options(google_cloud_cpp_mocks
+                       INTERFACE ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
+add_library(google-cloud-cpp::mocks ALIAS google_cloud_cpp_mocks)
+
+create_bazel_config(google_cloud_cpp_mocks YEAR "2022")
+
+google_cloud_cpp_install_headers("google_cloud_cpp_mocks"
```

Comment:
FYI, I think this will install the `.h` file, but we are not installing the targets (see #5782).  Can you check that the header installs correctly (maybe look at the installed files in one of the `demo-*` builds?)

Reply:
I run
```
ci/cloudbuild.sh -t demo-fedora-pr -s
# (inside docker shell now)

ci/cloudbuild.sh --local --build demo-install
# (build runs)

ls /h/google-cloud-cpp-installed/include/google/cloud/mocks/
# Output:
#    mock_stream_range.h
```

And if I run:
```
grep -rl mocks /h/google-cloud-cpp-installed/lib64/cmake/
# Output:
#    /h/google-cloud-cpp-installed/lib64/cmake/google_cloud_cpp_pubsub/pubsub-targets.cmake
```

I take it that #5782 involves getting the other mock libraries to show up.

Reply:
Maybe done? But does it need a pkgconfig thing like we do with `rest_internal`?

Should it have its own target separate from `google_cloud_cpp_common-targets` ?

Reply:
pkg-config has been left for a follow up PR.

I did move the targets out of `google_cloud_cpp_common-targets.cmake` and into a new `google_cloud_cpp_mocks-targets.cmake`. I think we want to do this to avoid installing the mock targets into the `google_cloud_cpp_runtime` component.

PTAL, because your original review was from a lifetime ago

Reply:
There is no `.pc` file, but there is no `-config.cmake` file either.  Is that intentional?